### PR TITLE
[Console-UI] Add status as filter in toolbar for address space list 

### PR DESCRIPTION
**Pull request by drcoolsanjeev**
### Type of change

- Enhancement / new feature
- Refactoring

### Description

- Add functionality to support filter address space by status
- Include terminating status in status filter dropdown for address
image

image

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by anukritijha** [comment]
@drcoolsanjeev Can you please update the template?

File: console/console-init/ui/mock-console-server/mock-console-server.js [hunk 1]
```diff
@@ -527,28 +595,35 @@ function createAddressSpace(as) {
...
       type: as.spec.type,
       endpoints: as.spec.endpoints ? as.spec.endpoints : createDefaultEndpoints(),
       authenticationService: {
-        name: as.spec.authenticationService ? as.spec.authenticationService.name : null
+        name: as.spec.authenticationService
+          ? as.spec.authenticationService.name
+          : null
       }
     },
-    status: null
```
Comment (drcoolsanjeev) [inline console/console-init/ui/mock-console-server/mock-console-server.js:551]: @k-wall I made a small change in the mock server as it is not matching with exact behaviour of real backend.

File: console/console-init/ui/mock-console-server/mock-console-server.js [hunk 2]
```diff
@@ -1058,34 +1188,50 @@ function createAddress(addr, addressSpaceName) {
       type: addr.spec.type,
       topic: addr.spec.topic
     },
-    status: null,
+    status: {
+      phase: "",
+      messages: []
+    }
```
Comment (drcoolsanjeev) [inline console/console-init/ui/mock-console-server/mock-console-server.js:1194]: @k-wall I made a small change in the mock server as it is not matching with exact behaviour of real backend.

File: console/console-init/ui/mock-console-server/mock-console-server.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ function calcLowerUpper(offset, first, len) {
   if (first !== undefined && first > 0) {
     upper = Math.min(lower + first, len);
   }
-  return {lower, upper};
+  return { lower, upper };
```
Comment (k-wall) [inline console/console-init/ui/mock-console-server/mock-console-server.js:26]: Combining the reformatting of the file with functional change is unhelpful to reviewers.  Can you revert the formatting changes?  If you want to reformat pls. submit it as a separate commit.,

File: console/console-init/ui/src/graphql-module/queries/Address.ts [hunk 4]
```diff
@@ -64,10 +64,18 @@ const ALL_ADDRESS_FOR_ADDRESS_SPACE_FILTER = (
...
-      { value: statusValue, isExact: true }
-    ]);
+    if (statusValue !== "Pending") {
+      filter += generateFilterPattern("status.phase", [
+        { value: statusValue, isExact: true }
+      ]);
+    } else {
+      filter += generateFilterPattern("status.phase", [
+        { value: statusValue, isExact: true },
+        { value: "", isExact: true }
```
Comment (ajaypratap003) [inline console/console-init/ui/src/graphql-module/queries/Address.ts:76]: is there any reason for adding second object with black value ?

File: console/console-init/ui/src/graphql-module/queries/AddressSpace.ts [hunk 5]
```diff
@@ -51,6 +52,20 @@ const ALL_ADDRESS_SPACES_FILTER = (
...
+  //filter tye
+  if (filterStatus) {
+    if (filterStatus !== "Pending") {
+      filter += generateFilterPattern("status.phase", [
+        { value: filterStatus, isExact: true }
+      ]);
+    } else {
+      filter += generateFilterPattern("status.phase", [
+        { value: filterStatus, isExact: true },
+        { value: "", isExact: true }
```
Comment (ajaypratap003) [inline console/console-init/ui/src/graphql-module/queries/AddressSpace.ts:65]: is there any reason for adding second object with black value ?

File: console/console-init/ui/src/graphql-module/queries/Address.ts (hunk 4 above)
Comment (drcoolsanjeev) [reply console/console-init/ui/src/graphql-module/queries/Address.ts:76]: Yes there is. When controller didn't pick the address or address-space for creation, the server assgin status.phase as empty string and in the UI we are displaying the value as Pending.

**Comment by drcoolsanjeev** [comment]
@enmasse-ci run tests testcase=FirefoxConsoleTest

**Comment by drcoolsanjeev** [comment]
@enmasse-ci run tests testcase=FirefoxConsoleTest

**Comment by drcoolsanjeev** [comment]
@enmasse-ci run tests testcase=isolated.web.FirefoxConsoleTest#testAddressSpaceSnippetStandard
